### PR TITLE
Enabled change path rule linter + added allow list

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

This adds the job-change-path linter that verifies dynamic jobs have `changes: paths: ...` rules to the pre check.
In addition, an allow list of the current failing jobs is added.

[Example job](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/568518547)

### CODE REVIEW

File: tasks/linter.py [hunk 1]
```diff
@@ -423,10 +423,113 @@ def update_go(_):
...
+        'new-e2e-agent-platform-step-by-step-suse-a6-x86_64',
+        'new-e2e-agent-platform-step-by-step-suse-a7-arm64',
+        'new-e2e-agent-platform-step-by-step-suse-a7-x86_64',
+        'new-e2e-agent-platform-step-by-step-ubuntu-a6-arm64',
+        'new-e2e-agent-platform-step-by-step-ubuntu-a6-x86_64',
+        'new-e2e-agent-platform-step-by-step-ubuntu-a7-arm64',
+        'new-e2e-agent-platform-step-by-step-ubuntu-a7-x86_64',
+        'new-e2e-agent-shared-components',
+        'new-e2e-agent-subcommands',
+        'new-e2e-aml',
```
Comment (chouetz) [inline tasks/linter.py:511]: If this list is for the jobs which are allowed to have a `changes:path` NOT defined them I'm surprised to find this, as they define their own paths [here](https://github.com/DataDog/datadog-agent/blob/main/.gitlab-ci.yml#L897) and it's properly referenced [here](https://github.com/DataDog/datadog-agent/blob/main/.gitlab/e2e/e2e.yml#L277)
It's confirmed [here](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/568518547#L65) you don't detect this test has having change paths defined, so could you please double check?
Thanks

Comment (CelianR) [reply tasks/linter.py]: Screenshot 2024-07-11 at 14 29 53

I updated my script, these jobs are now correctly handled. It was due to references that create lists of lists

File: tasks/linter.py [hunk 2]
```diff
@@ -450,20 +542,44 @@ def contains_valid_change_rule(rule):
 
     # Verify that all tests contain a change path rule
     tests_without_change_path = defaultdict(list)
+    tests_without_change_path_allowed = defaultdict(list)
     for test, filepath in tests:
+        # Not a job
```
Comment (chouetz) [inline tasks/linter.py:547]: what could it be?

**Review by chouetz — APPROVED** [review]

File: tasks/linter.py (hunk 2 above)
Comment (CelianR) [reply tasks/linter.py:547]: Removed :)

**Pull request merged**